pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -2853,7 +2855,41 @@ public boolean isOldestMessageExpired(ManagedCursor cursor, int messageTTLInSeco
                         (int) (messageTTLInSeconds * MESSAGE_EXPIRY_THRESHOLD), entryTimestamp);
             }
         } catch (Exception e) {
-            log.warn("[{}] Error while getting the oldest message", topic, e);
+            if (brokerService.pulsar().getConfiguration().isAutoSkipNonRecoverableData()
```

Comment:
What about simply set `isOldestMessageExpired` as `true` and let it go through if any exceptions thrown?  Because `expiryMonitor.expireMessages(messageTTLInSeconds)` will handle the non-recoverable data issue

Reply:
> What about simply set isOldestMessageExpired as true and let it go through if any exceptions thrown?
Good idea.  But I think we  should  alose consider the exception type. Only NonRecoverableLedgerException can reture true. WDYT?